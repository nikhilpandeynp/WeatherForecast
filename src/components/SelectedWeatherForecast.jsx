import React, { Component } from 'react';
import { fetchWeatherInformation } from '../WeatherForecastAPI';
import WeatherForecast from './WeatherForecast';
import ReactLoading from 'react-loading';

export default class SelectedWeatherForecast extends Component {
    constructor(props) {
        super(props);
        this.state = {
            lat: this.props.lat,
            long: this.props.long,
            weatherDataFetched: false
        };
    }

    static getDerivedStateFromProps(nextProps, prevState) {
        if (prevState.lat === nextProps.lat && prevState.long === nextProps.long) {
            return null;
        } else {
            return {
                weatherDataFetched: false,
                lat: nextProps.lat,
                long: nextProps.long
            };
        }
    }

    fetchWeatherInformationFromAPI(lat, long) {
        fetchWeatherInformation(lat, long).then((weatherForecast) => {
            this.setState({
                weatherForecast: weatherForecast.slice(0, 5),
                weatherDataFetched: true
            });
        }).catch((err) => {
            throw new Error(err);
        });
    }

    render() {
        if (this.state.weatherDataFetched) {
            return <WeatherForecast weatherForecast={this.state.weatherForecast} />
        } else {
            this.fetchWeatherInformationFromAPI(this.state.lat, this.state.long);
            return <ReactLoading type="spin" color="#0000FF" height={67} width={37} />
        }
    }
}