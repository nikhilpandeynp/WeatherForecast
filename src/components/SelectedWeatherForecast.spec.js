import SelectedWeatherForecast from './SelectedWeatherForecast';
import renderer from 'react-test-renderer';
import React from 'react';

jest.mock('./WeatherForecast');
jest.mock('react-loading');
jest.mock('../WeatherForecastAPI');

it('should show react loading when data is not fetched', () => {
    let props = {
        lat: 1,
        long: 2
    };
    let component = renderer.create(<SelectedWeatherForecast {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});