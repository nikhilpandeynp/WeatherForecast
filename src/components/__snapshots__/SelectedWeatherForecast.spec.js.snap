// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show react loading when data is not fetched 1`] = `null`;
